### ISSUE TITLE
incorrect compile error

### ISSUE THREAD

**Issue by ghosttie** [google/closure-compiler#933]
I have a property defined as:

```javascript
/**
 * @type {test.DisplayObject|Array<test.DisplayObject>|Array<test.TableRow>}
 */
this.value;
```

but the compiler seems to be confused about it:

    found   : (Array|null|test.DisplayObject)
    required: (null|test.DisplayObject)
            var /** test.DisplayObject|Array<test.DisplayObject>|Array<test.TableRow> */ value = new test.Result().value;
                                                                                                 ^

It seems to have collapsed the typed arrays to Array for the found value and the arrays are completely missing from the variable's type.

I have created a [gist](https://gist.github.com/ghosttie/86a576ad5681baa70747) that reproduces the problem.

**Comment by MatrixFrog** [comment]
I just pushed a new version of the debugger service, and I'm not seeing this. Maybe you're on an older compiler version? http://closure-compiler-debugger.appspot.com/#input0%3D%252F**%250A%2520*%2520%2540constructor%250A%2520*%2520%2540export%250A%2520*%252F%250Avar%2520test%2520%253D%2520function()%2520%257B%257D%253B%250A%2520%250Atest.prototype.doTest1%2520%253D%2520function()%2520%257B%250A%2509var%2520%252F**%2520test.DisplayObject%257CArray%253Ctest.DisplayObject%253E%257CArray%253Ctest.TableRow%253E%2520*%252F%2520value%2520%253D%2520new%2520test.Result().value%253B%250A%257D%253B%250A%250A%252F**%250A%2520*%2520%2540constructor%250A%2520*%2520%2540struct%250A%2520*%252F%250Atest.DisplayObject%2520%253D%2520function()%2520%257B%257D%253B%250A%2520%250A%252F**%250A%2520*%2520%2540constructor%250A%2520*%2520%2540struct%250A%2520*%252F%250Atest.TableRow%2520%253D%2520function()%2520%257B%257D%253B%250A%2520%250A%252F**%250A%2520*%2520%2540constructor%250A%2520*%2520%2540struct%250A%2520*%252F%250Atest.Result%2520%253D%2520function()%2520%257B%250A%2509%252F**%250A%2509%2520*%2520%2540type%2520%257Btest.DisplayObject%257CArray%253Ctest.DisplayObject%253E%257CArray%253Ctest.TableRow%253E%257D%250A%2509%2520*%252F%250A%2509this.value%253B%250A%257D%253B%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26MISSING_PROPERTIES%3D1

**Comment by ghosttie** [comment]
I'm using the latest version of the compiler (v20150505).

I'm not sure what to do with your link - I tried pasting my reproducer code into the text boxes but it says

    input0:1: ERROR - variable goog is undeclared
    goog.provide('test');
    ^

**Comment by MatrixFrog** [comment]
My link is the same, just without the goog.provide (since the debugger service doesn't have the Closure Library built in)

What is the full set of command line flags you're using?

**Comment by ghosttie** [comment]
java -jar compiler.jar ^
			--js=_closure-library ^
			--js=c:\temp\test.js ^
			--compilation_level=ADVANCED_OPTIMIZATIONS ^
			--warning_level=VERBOSE ^
			--jscomp_error=accessControls ^
			--jscomp_error=checkRegExp ^
			--jscomp_error=const ^
			--jscomp_error=constantProperty ^
			--jscomp_error=strictModuleDepCheck ^
			--jscomp_error=visibility ^
			--jscomp_error=missingProperties ^
			--only_closure_dependencies ^
			--closure_entry_point=test ^
			--js_output_file=c:\temp\test_compiled.js ^
			--generate_exports

**Comment by Dominator008** [comment]
@ghosttie Are you still getting that error?
While I couldn't reproduce the typing error, I'm getting a crash shown in #1393.

**Comment by Dominator008** [comment]
Can't reproduce the original error in the latest release (v20151216), closing this.

**Comment by ghosttie** [comment]
No, I can no longer reproduce it, thanks.
